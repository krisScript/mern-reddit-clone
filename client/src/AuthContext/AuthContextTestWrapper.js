import {AuthContextData} from './AuthContext'
import {BrowserRouter} from 'react-router-dom'
import React from 'react'
const AuthContextTestWrapper = ({authState,children,updateUserDataReducer}) => {
    return (
       <AuthContextData.Provider value={{authState,updateUserDataReducer}}>
        <BrowserRouter>
          {children}
          </BrowserRouter>
        </AuthContextData.Provider>
    )
}
export default AuthContextTestWrapper 