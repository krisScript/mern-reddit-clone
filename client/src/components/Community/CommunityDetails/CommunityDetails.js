import React,{useContext,Fragment} from 'react'
import SubscribeButton from '../../SubscribeButton/SubscribeButton'
import { Link} from "react-router-dom";
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import useAuthContext from '../../../hooks/useAuthContext/useAuthContext'
import PropTypes from 'prop-types';
import communityType from '../../PropTypes/communityType'
import {
 faShieldAlt
} from '@fortawesome/free-solid-svg-icons';
library.add(faShieldAlt);
const CommunityDetails = ({community}) => {
    const {userId} = useAuthContext()
    const {name,icon,subscribers,_id,creator} = community
    return (
        <>
        <div className="community-details">
        <div className="community-details-header">
        <h3 className="community-details-title">Community Details </h3>
        {creator === userId ? <Link to={`/mod-tools/${community._id}`}><button className="button button-icon"><FontAwesomeIcon icon="shield-alt" /><span>Mod Tools</span>
        </button></Link> : <></> }
        </div>
        <div className="community-banner">
            <div className="community-logo">
            <img src={`http://localhost:8080/images/${icon}`}  alt={`${name}`} data-testid="community-logo"/>
            </div>
            </div>
            <h1 className="community-name">{name}</h1>
            <h2 className="community-subscribers">Subscribers {subscribers}</h2>
            <div className="buttons-container">
            <SubscribeButton id={_id}/>
            <button className="button">
                <Link data-testid="create-post"  to={`/create-post/${_id}`}>
                Create Post
                </Link>
            </button>
            </div>
        </div>
        </>
    )
}

CommunityDetails.propTypes = {
    community: communityType.isRequired,
  };
export default CommunityDetails 