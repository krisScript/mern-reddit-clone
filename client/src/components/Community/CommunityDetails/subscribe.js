const subscribe = async (communityId,token) => {
     const response  =  await fetch(`http://localhost:8080/subscribe/${communityId}`, {
        method: 'PUT',
        headers: {
          Authorization: 'Bearer ' + token,
          'Content-Type': 'application/json'
        },
      });

      const responseData = await response.json()

      return responseData
}
export default subscribe