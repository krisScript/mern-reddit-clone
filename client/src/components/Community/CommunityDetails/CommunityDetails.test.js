import React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import CommunityDetails from './CommunityDetails';
import AuthContextTestWrapper from '../../../AuthContext/AuthContextTestWrapper'
const updateUserDataReducer = jest.fn()
describe('<CommunityDetails/>',() => {
    
    const community = {
      name:'test',
      icon:'jest',
      subscribers:3000,
      _id:'121d2',
      creator:'asdasdca'
    }
    const deletePostElement  = jest.fn()

    const authState = {
      isAuth:true,userId:community.creator,token:'sadsda',userData:{communities:[]}
    }
    const { container,getByText,getByTestId } = render (<CommunityDetails  community={community}  />,{wrapper: ({ children }) => (
      <AuthContextTestWrapper authState={authState} children={children} />
      )})
      it('snapshot',() => {
        expect(container).toMatchSnapshot()
    })
    it('should have element image with test id "community-logo"', async () => {
        const communityLogo =  getByTestId(`community-logo`)
        expect(communityLogo).toBeTruthy()
        expect(communityLogo.src).toMatch(`http://localhost:8080/images/${community.icon}`)
        expect(communityLogo.alt).toMatch(community.name)
      });
      it('should have element with "community.name" text', async () => {
        const communityName =  getByText(`${community.name}`)
        expect(communityName.className).toMatch('community-name')
        expect(communityName).toBeTruthy;
      });
      it('should have element with "Subscribers community.subscribers" text', async () => {
        const communitySubscribers =  getByText(`Subscribers ${community.subscribers}`)
        expect(communitySubscribers.className).toMatch('community-subscribers')
        expect(communitySubscribers).toBeTruthy;
      });
      it('should have element with "Create Post" text', async () => {
        const createPostButton =  getByText('Create Post')
        expect(createPostButton).toBeTruthy;
      });
      it('mode tools link should exist', async () => {
        const modToolsLink =  getByText('Mod Tools')
        expect(modToolsLink).toBeTruthy;
      });
      
})
