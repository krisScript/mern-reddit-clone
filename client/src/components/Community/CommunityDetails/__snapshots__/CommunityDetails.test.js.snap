// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommunityDetails/> snapshot 1`] = `
<div>
  <div
    class="community-details"
  >
    <div
      class="community-details-header"
    >
      <h3
        class="community-details-title"
      >
        Community Details 
      </h3>
      <a
        href="/mod-tools/121d2"
      >
        <button
          class="button post-info-button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-shield-alt fa-w-16 "
            data-icon="shield-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M466.5 83.7l-192-80a48.15 48.15 0 0 0-36.9 0l-192 80C27.7 91.1 16 108.6 16 128c0 198.5 114.5 335.7 221.5 380.3 11.8 4.9 25.1 4.9 36.9 0C360.1 472.6 496 349.3 496 128c0-19.4-11.7-36.9-29.5-44.3zM256.1 446.3l-.1-381 175.9 73.3c-3.3 151.4-82.1 261.1-175.8 307.7z"
              fill="currentColor"
            />
          </svg>
          <span>
            Mod Tools
          </span>
        </button>
      </a>
    </div>
    <div
      class="community-banner"
    >
      <div
        class="community-logo"
      >
        <img
          alt="test"
          data-testid="community-logo"
          src="http://localhost:8080/images/jest"
        />
      </div>
    </div>
    <h1
      class="community-name"
    >
      test
    </h1>
    <h2
      class="community-subscribers"
    >
      Subscribers 
      3000
    </h2>
    <div
      class="buttons-container"
    >
      <button
        class="button"
      >
        Subscribe
      </button>
      <button
        class="button"
      >
        <a
          data-testid="create-post"
          href="/create-post/121d2"
        >
          Create Post
        </a>
      </button>
    </div>
  </div>
</div>
`;
